packages/shared/lib/services/connection.service.ts

```diff
@@ -702,6 +702,7 @@ class ConnectionService {
                     });
                 }
 
+                // TODO: this leak credentials to the logs
                 const errorWithPayload = new NangoError(error.type, connection);
 
                 return Err(errorWithPayload);
```

Comment:
This return is not nice, it's leaking credentials and also force the API to return a 400 which is not true